src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceMetadata.java

```diff
@@ -729,7 +729,12 @@ MetadataResponse getMetadataResponse() throws SQLException {
                         response.code(), response.headers().get(Constants.TRACE_ID));
                 HttpHelper.handleErrorResponse(response, Constants.MESSAGE);
             }
-            return HttpHelper.handleSuccessResponse(response, MetadataResponse.class, true);
+            StringBuilder cacheKey = new StringBuilder(response.request().url().toString());
```

Comment:
@KaviarasuSakthivadivel : If the client logs out, and then logs in with a different tenant, wouldn't it still load from the same cache, if the dataspace is same

Reply:
I have changed the code! Can you please review it? @vish689